src/app/cli/src/app.rs

```diff
@@ -128,17 +128,22 @@ pub async fn run(workspace_layout: WorkspaceLayout, args: cli::Cli) -> Result<()
             tenancy_config,
             current_account.to_current_account_subject(),
         );
+        let is_e2e_testing = args.e2e_output_data_path.is_some();
 
         let mut base_catalog_builder = configure_base_catalog(
             &workspace_layout,
             tenancy_config,
             args.system_time.map(Into::into),
-            args.e2e_output_data_path.is_some(),
+            is_e2e_testing,
         );
 
-        if workspace_svc.is_in_workspace() {
-            base_catalog_builder.add::<DatasetEntryIndexer>();
-        }
+        base_catalog_builder.add_builder(
+            kamu_datasets_services::DatasetEntryIndexer::builder()
+                .with_is_in_workspace(workspace_svc.is_in_workspace()),
+        );
+        // The indexer has no other interfaces
+        base_catalog_builder
```

Comment:
Yes, since other entities in the `InitOnStartup` context depend on it -- so I had to add a pass inside `is_in_workspace`.

But you're right, it should be more explicit -- added comment